app_flutter/lib/agent_list.dart

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import 'package:cocoon_service/protos.dart' show Agent;
+
+import 'agent_health_details.dart';
+import 'agent_tile.dart';
+import 'state/agent.dart';
+
+class AgentList extends StatelessWidget {
+  const AgentList(
+      {this.agents,
+      this.agentState,
+      @visibleForTesting this.insertKeys = false});
+
+  final List<Agent> agents;
+
+  final AgentState agentState;
+
+  /// When true, will set a key for the [AgentTile] that is composed of its
+  /// position in the list and the agent id.
+  @visibleForTesting
+  final bool insertKeys;
+
+  @override
+  Widget build(BuildContext context) {
+    if (agents.isEmpty) {
+      return const CircularProgressIndicator();
+    }
+
+    final List<FullAgent> fullAgents = agents
```

Comment:
Could you sort the agents where you download them, to avoid re-sorting unnecessarily?

Reply:
I plan to update the backend in a followup PR for this (https://github.com/flutter/flutter/issues/48249).

Reply:
I plan to update the backend in a followup PR for this (https://github.com/flutter/flutter/issues/48249).